[TASK-397] BaseEntity 생성

blacklabf opened the pull request:
[![workerB](https://img.shields.io/endpoint?url=https%3A%2F%2Fworkerb.linearb.io%2Fv2%2Fbadge%2Fprivate%2FU2FsdGVkX1XRTuUEfp542M5MZzCaQrSmpc1QjY%2Fcollaboration.svg%3FcacheSeconds%3D60)](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=z1wT7-0)
## Description
모든 테이블에 공통적으로 적용되는 생성일시, 수정일시, 데이터삭제여부 속성을 관리하는 BaseEntity 생성
[Task-397](https://www.notion.so/BaseEntity-35b12353fa1c4f46baf011f006a8dc50?pvs=4)<br/>

## 결과
* 모든 테이블에 적용
![모든테이블에 적용](https://github.com/Speechpeach/speech-backend/assets/108640460/398ba67c-a5fe-4382-b8e7-39dab6e8a7c3)

* 생성일시, 수정일시, 데이터 삭제여부가 들어가는지 테스트코드로 확인했습니다. 

## More
@Kim-AYoung  추후에 BaseEntity에 생성자, 수정자 필드 추가부탁드립니다! 

## Issue
#8

blacklabf committed:
setting: BaseEntity 생성

1. 모든 테이블에 공통적으로 적용되는 생성자, 수정자, 생성일시, 수정일시, 데이터삭제여부 속성을 관리하는 BaseTimeEntity와 BaseEntity 생성

blacklabf committed:
setting: BaseEntity 생성

1. 모든 클래스에 공통 속성인 생성일시, 수정일시, 데이터삭제여부 관리하는 BaseEntity 생성

blacklabf committed:
setting: BaseEntity 생성

1. 모든 클래스에 공통 속성인 생성일시, 수정일시, 데이터삭제여부 관리하는 BaseEntity 생성
2. 데이터 제대로 들어가는지 테스트 완료

Kim-AYoung commented on src/main/java/com/speechpeach/speech/global/config/AuditConfig.java:
```diff
@@ -0,0 +1,10 @@
+
+@Configuration
+@EnableJpaAuditing
+public class AuditConfig {
```
👍 커스텀 Config 클래스로 분리하니 깔끔해서 좋네요!

Kim-AYoung commented on src/test/java/com/speechpeach/speech/global/entity/BaseEntityTest.java:
```diff
@@ -0,0 +1,47 @@
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
```
⚠ 저도 로컬 mysql DB를 사용하기위해 이 어노테이션을 추가해놨었는데, github test CI에서 제 로컬 DB를 연결하지 못해 불통과가 뜨더라구요
이 어노테이션을 제거하니 `@DataJpaTest`가 기본적으로 사용하는 인메모리 DB가 연결되어 통과되었습니다!
(추가적으로 `build.gradle`에 h2 의존성(`testRuntimeOnly 'com.h2database:h2'`)도 추가하셔야합니다!)

blacklabf replied:
h2 db 연결해서 사용하니까 잘 됩니다! 감사합니다! 
그리고 이모지로 구분해서 코멘트 달아주셔서 가독성이 좋네요! 감사합니다!

Kim-AYoung commented on src/test/java/com/speechpeach/speech/global/entity/BaseEntityTest.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Test
+    @DisplayName("생성일시, 수정일시, 데이터삭제여부 잘 들어가는지 확인")
+    void BaseEntitySetTest() throws NoSuchFieldException, IllegalAccessException{
```
✅ 자바 네이밍 컨벤션에 맞게 테스트 메소드명이 카멜케이스로 작성되면 좋을 것 같습니다!

Kim-AYoung replied:
➕ 추후 `BaseEntity`를 상속한 클래스가 만들어진다면, 해당 클래스가 repository에 저장되고 업데이트 될 때 createdDate, modifiedDate 필드가 null이 아닌지 체크하는 테스트 코드가 추가되면 좋을 것 같습니다!

blacklabf replied:
리뷰 자세하게 필요한 부분 많이 해주셔서 감사합니다! 
잘 배워갑니다! 다시 수정해서 pr 하겠습니다!

Kim-AYoung commented:
> BaseEntity에 생성자, 수정자 필드 추가부탁드립니다!

이후 토큰 기능을 구현하면, 추가해놓도록 하겠습니다!

blacklabf committed:
setting: BaseEntity 생성

1. 모든 클래스에 공통 속성인 생성일시, 수정일시, 데이터삭제여부 관리하는 BaseEntity 생성
2. 각 컬럼에 값 제대로 들어가는지 test 성공
3. test db를 인메모리 h2 db 연결로 수정

Kim-AYoung reviewed (approved):
👍 LGTM

beautiflow reviewed (approved):
LGTM👏

blacklabf committed:
Merge branch 'develop' of https://github.com/Speechpeach/speech-backend into feature/task-394-video_skeleton
